proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1790 @@
+    the nearest integer.
+
+  * Let `green1` and `green2` be the green channels of `color1` and `color2`
+    respectively.
+
+  * Let `green` be the result of rounding `green1 * weight1 + green2 * weight2`
+    to the nearest integer.
+
+  * Let `blue1` and `blue2` be the blue channels of `color1` and `color2`
+    respectively.
+
+  * Let `blue` be the result of rounding `blue1 * weight1 + blue2 * weight2` to
+    the nearest integer.
+
+  * Let `alpha` be `alpha1 * weight-scale + alpha2 * (1 - weight-scale)`.
+
+  * Return a [legacy color] in the `rgb` space, with the given `red`, `green`,
+    and `blue` channels, and `alpha` value.
+
+* Otherwise:
+
+  * If `$method` is not a [color interpolation method][color-method], throw an
+    error.
+
+  * Let `space` be the [known color space] specified in `$method`.
+
+* If `space` is unquoted and `space == hsl` or `space == hwb`:
+
+  * Let `color1` and `color2` be the result of [converting] and
+    [gamut mapping] `$color1` and `$color2` respectively into `space`.
+
+  > These color spaces are unable to express colors outside the `srgb` gamut.
+
+* Otherwise, let `color1` and `color2` be the result of [converting]
+  `$color1` and `$color2` respectively into `space`.
+
+* Return the result of [interpolating] between `color1` and `color2` with the
+  specified `$weight` and `$method`.
+
+### `color.change()`
+
+```
+change($color, $args...)
+```
+
+This function is also available as a global function named `change-color()`.
+
+* If `$color` is not a color, throw an error.
+
+* If any item in `$args` is not a keyword argument, throw an error.
+
+* Let `space` be `$color`'s color space.
+
+* If the keyword argument `$space` is specified in `$args`:
+
+    * If `$space` is not a [known color space], and `$space != space`, throw an
+      error.
+
+    * Set `space` to the value of `$space`.
+
+    * Let `color` be the result of [converting] `$color` to `space`.
+
+* Otherwise, let `color` be the value of `$color`.
+
+* Let `legacy` be `true` if `$color` is a legacy color and `space` is a
+  legacy color space, and `false` otherwise.
+
+* Let `alpha` be `color`'s alpha property.
+
+* If the keyword argument `$alpha` is specified in `$args`:
+
+    * Set `alpha` to the result of [percent-converting] `$alpha`, and clamping
+      it between 0 and 1 (inclusive).
+
+* Let `channel-args` be the remaining keyword arguments in `$args`, not
+  including `$space` or `$alpha` arguments.
+
+* Let `channels` be a list of the `color`'s channels.
+
+* For each keyword `key` and value `new` in `channel-args`:
+
+  * If `key` is a string in the format `channel<integer>`, set `key` to the
+    value of `<integer>`.
+
+    > This allows e.g. `color.change($color, $channel1: 0.25)` for changing
+    > color channels in unknown color spaces.
+
+  * If `key` is not the name or index of a channel in `channels`, throw an
+    error.
+
+  * Set the corresponding `channel` in `channels` to `new`.
```

Comment:
We should verify the value of `new` is valid. Same for `adjust()` and `scale()`.

Reply:
This is handled by the normalization step, after we've made all the requested changes.